src/utils/getUrlByRequestOptions.ts

```diff
@@ -16,6 +16,8 @@ export function getUrlByRequestOptions(
 
   debug('creating URL from options:', options)
 
+  options.protocol = options.protocol || (options.agent && (options.agent as RequestOptions).protocol) || undefined
```

Comment:
Can we use optional chaining for a shorter reference?

```js
options.protocol = options.protocol || options.agent?.protocol || undefined
```

Does this raise a type violation?

Reply:
I see that something similar is also for port. Maybe we should do the same ?

https://github.com/nodejs/node/blob/d84f1312915fe45fe0febe888db692c74894c382/lib/_http_client.js#L158-L161

Reply:
Great point, @marcosvega91. Added this in e6e4edf7588297133666ab6f4c300cdac4f85d53